include/zephyr/secure_storage/its.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2024
```

Comment:
Got the point. I think having these headers publicly available will cause people using them directly instead of psa* ones that are not in `include/zephyr`.  Maybe we should move them to `subsys/secure_storage/include` and only include them in the include path if the application sets `CONFIG_SECURE_STORAGE_ITS_IMPLEMENTATION_CUSTOM=y`. 

The way to use it is through PSA APIs so we need to make it clear, having a custom implementation is not the common use case so we don't need to promote these headers unless someone explicitly needs them.

What you think ?

Reply:
It makes sense.
I am working on adding CMake logic that will make `secure_storage` header files available only if the relevant `*_CUSTOM` Kconfig options are enabled.
It's a bit more complex than just conditionally adding an include path because there are different levels of granularity, but it's feasible.

Reply:
Solved as part of https://github.com/zephyrproject-rtos/zephyr/compare/37d6ae39bd509ee1f7a40f387f6a87a0fdfa3e4e..43bba9bf84ca53f3d134118b5d99f623602ce474#diff-ac1f1e035e1eb038ec993472202e54c6b07cd1d23aa0346426936e282c5ead8d.